modulemd/v2/modulemd-subdocument-info.c

```diff
@@ -0,0 +1,300 @@
+  modulemd_subdocument_info_set_gerror (
+    s, modulemd_subdocument_info_get_gerror (self));
+  modulemd_subdocument_info_set_yaml (
+    s, modulemd_subdocument_info_get_yaml (self));
+
+  return g_steal_pointer (&s);
+}
+
+
+static void
+modulemd_subdocument_info_finalize (GObject *object)
+{
+  ModulemdSubdocumentInfo *self = (ModulemdSubdocumentInfo *)object;
+
+  g_clear_pointer (&self->error, g_error_free);
+  g_clear_pointer (&self->contents, g_free);
+
+  G_OBJECT_CLASS (modulemd_subdocument_info_parent_class)->finalize (object);
+}
+
+
+void
+modulemd_subdocument_info_set_yaml (ModulemdSubdocumentInfo *self,
+                                    const gchar *yaml)
+{
+  g_return_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self));
+
+  printf ("Setting YAML: %s\n", yaml);
+
+  g_clear_pointer (&self->contents, g_free);
+  self->contents = g_strdup (yaml);
+}
+
+
+const gchar *
+modulemd_subdocument_info_get_yaml (ModulemdSubdocumentInfo *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self), NULL);
+
+  return self->contents;
+}
+
+
+void
+modulemd_subdocument_info_set_gerror (ModulemdSubdocumentInfo *self,
+                                      const GError *error)
+{
+  g_return_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self));
+
+  g_clear_pointer (&self->error, g_error_free);
+  if (error)
+    {
+      fprintf (stderr, "Setting subdoc error: %s\n", error->message);
+      self->error = g_error_copy (error);
+    }
+  else
+    {
+      fprintf (stderr, "Clearing subdoc error\n");
+      self->error = NULL;
+    }
+}
+
+
+const GError *
+modulemd_subdocument_info_get_gerror (ModulemdSubdocumentInfo *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self), NULL);
+
+  return self->error;
+}
+
+
+void
+modulemd_subdocument_info_set_doctype (ModulemdSubdocumentInfo *self,
+                                       enum ModulemdYamlDocumentType doctype)
+{
+  g_return_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self));
+
+  self->doctype = doctype;
+}
+
+
+enum ModulemdYamlDocumentType
+modulemd_subdocument_info_get_doctype (ModulemdSubdocumentInfo *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self),
+                        MODULEMD_YAML_DOC_UNKNOWN);
+
+  return self->doctype;
+}
+
+
+void
+modulemd_subdocument_info_set_mdversion (ModulemdSubdocumentInfo *self,
+                                         guint64 mdversion)
+{
+  g_return_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self));
+
+  self->mdversion = mdversion;
+}
+
+
+guint64
+modulemd_subdocument_info_get_mdversion (ModulemdSubdocumentInfo *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self), 0);
+
+  return self->mdversion;
+}
+
+
+gboolean
+modulemd_subdocument_info_get_data_parser (ModulemdSubdocumentInfo *self,
+                                           yaml_parser_t *parser,
+                                           GError **error)
+{
+  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self), FALSE);
```

Comment:
We should have a check for the validity of the GError pointer here.
```suggestion
  g_return_val_if_fail (MODULEMD_IS_SUBDOCUMENT_INFO (self), FALSE);
  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
```